lib/src/widgets/adw/view_switcher_tab.dart

```diff
@@ -5,59 +5,73 @@ class AdwViewSwitcherTab extends StatelessWidget {
   final ViewSwitcherData data;
   final ViewSwitcherStyle style;
   final bool isSelected;
+  final VoidCallback? onSelected;
 
   const AdwViewSwitcherTab({
     Key? key,
     required this.data,
-    required this.isSelected,
     required this.style,
+    this.isSelected = false,
+    this.onSelected,
   }) : super(key: key);
 
   @override
   Widget build(BuildContext context) {
-    final icon = Icon(
-      data.icon,
-      size: 18,
-    );
+    final isDesktop = style == ViewSwitcherStyle.desktop;
 
-    return style == ViewSwitcherStyle.desktop
-        ? Padding(
-            padding: const EdgeInsets.symmetric(horizontal: 24.0),
-            child: Row(
-              children: [
-                if (data.icon != null) icon,
-                if (data.icon != null && data.title != null)
-                  const SizedBox(width: 8),
-                if (data.title != null)
-                  Text(
-                    data.title!,
-                    style: Theme.of(context).textTheme.bodyText2?.copyWith(
-                          fontWeight:
-                              isSelected ? FontWeight.bold : FontWeight.normal,
-                        ),
-                  ),
-              ],
+    return AdwButton.flat(
+      constraints: isDesktop
+          ? const BoxConstraints(minWidth: 120, minHeight: 34, maxHeight: 36)
+          : const BoxConstraints(minWidth: 75),
+      margin: isDesktop
+          ? const EdgeInsets.symmetric(vertical: 6).copyWith(right: 3)
+          : EdgeInsets.zero,
+      onPressed: onSelected,
+      isActive: isSelected,
+      textStyle: TextStyle(
+        fontSize: isDesktop ? null : 11,
+      ),
+      borderRadius: isDesktop
+          ? const BorderRadius.all(Radius.circular(6.0))
+          : BorderRadius.zero,
+      child: _AdwViewSwitcherTabLayout(
+        isRow: isDesktop,
+        children: [
+          if (data.icon != null)
+            Icon(
+              data.icon,
+              size: 18,
```

Comment:
I'm not 100% sure about this values. Did you check whether the default icon size for these widgets is `18`?

Reply:
They are a little smaller but I think it looks OK.